mash/mash.go

```diff
@@ -0,0 +1,88 @@
+/*
+Package mash is for sketching sequence data to make it easier to search for and against.
+
+The package is named mash after the mash sketching algorithm, which is based on the MinHash algorithm.
+
+Mash: fast genome and metagenome distance estimation using MinHash.
+Ondov, B.D., Treangen, T.J., Melsted, P. et al.
+Genome Biol 17, 132 (2016).
+https://doi.org/10.1186/s13059-016-0997-x
+
+Mash Screen: high-throughput sequence containment estimation for genome discovery.
+Ondov, B., Starrett, G., Sappington, A. et al.
+Genome Biol 20, 232 (2019).
+https://doi.org/10.1186/s13059-019-1841-x
+
+The idea is that we can sketch a sequence of data, and then compare the sketch to other sketches to see how similar they are.
+This saves a bunch of computation time and memory, because we don't have to compare the entire sequence to another sequence.
+
+TTFN,
+Tim
+*/
+package mash
+
+import "github.com/spaolacci/murmur3"
+
+// sketch algorithm
+// slide a window of size k along the sequence
+// for each kmer in the window, hash it to a 32 or 64 bit number
+// keep the minimum hash value of all the kmers in the window up to a given sketch size
+// the sketch is a vector of the minimum hash values
+
+// what are mash's inputs and outputs?
+// inputs: a
+
+type Mash struct {
+	KmerSize   uint
+	SketchSize uint
+	Sketches   []uint32
+}
+
+func NewMash(kmerSize uint, sketchSize uint) *Mash {
```

Comment:
Also I'd avoid `uint` in size definitions to catch programmer errors early. Having sizes as uint can do more harm than good i.e:

```go
sketchSize := uint(len(sketchPool) - totalProcessed)
```
In the code above the user calculates the sketch size from a pair of `int`s. If the user is mistaken and the result of the calculation is negative then `uint` converts the negative number into a massively huge number which would cause a out of memory panic.

Compared to negative size panics the OOM panic can affect other goroutines running in the same program.